src/user/friendship.ts

```diff
@@ -70,6 +71,40 @@ export class Friendship extends Accessory implements Acceptable {
     return newFriendship
   }
 
+  /**
+   * Search a Friend by phone or weixin.
+   *
+   * The best practice is to search friend request once per minute.
+   * Remeber not to do this too frequently, or your account may be blocked.
+   *
+   * @param {FriendshipSearchCondition} condition - Search friend by phone or weixin.
+   * @returns {Promise<Contact>}
+   *
+   * @example
+   * const friend_phone = await bot.Friendship.search({phone: '13112341234'})
+   * const friend_weixin = await bot.Friendship.search({weixin: 'weixin_account'})
+   *
+   * console.log(`This is the new friend info searched by phone : ${friend_phone}`)
+   * await bot.Friendship.add(friend_phone, 'hello')
+   *
+   */
+  public static async search (
+    condition: FriendshipSearchCondition
+  ): Promise<Contact> {
+    log.verbose('Friendship', 'static search(%s)',
+      condition.phone || condition.weixin
+    )
+    if (condition.phone || condition.weixin) {
+      const id = condition.phone || condition.weixin || ''
```

Comment:
We don't need the `|| ''` since there is a check on the `condition.phone` and `condition.weixin`

Reply:
Of course, but without empty string the id will be regard as undefined type, and then we have to use `id!`, which way is better? Or any other better way?